docs/contributor/adr/0010-geoblocking-draft.md

```diff
@@ -0,0 +1,387 @@
+|                    |                                                         |                                                         |
+| Caching service    | Extensibility (may cache more things in future)         | Complex - another component to take care                |
+|                    | Works well with restart / upgrade / scaling             |                                                         |
+|                    | No downtime if geoblocking service doesn't work         |                                                         |
+|                    | Less load on geoblocking service                        |                                                         |
+|                    |                                                         |                                                         |
+| Config map         | Still simple                                            | Configmap capacity allows max ~50000 IP ranges          |
+|                    | Works well with restart / upgrade / scaling             | IP list integrity not guaranteed                        |
+|                    | No downtime if geoblocking service doesn't work         |                                                         |
+|                    | Less load on geoblocking service                        |                                                         |
+|                    |                                                         |                                                         |
+| Ephemeral volume   | Still simple                                            | Every container causes load on geoblocking service      |
+|                    | No downtime if geoblocking service doesn't work         | Doesn't help in case of restart / upgrade / scaling     |
+|                    |                                                         |                                                         |
+| Persistent volume  | No downtime if geoblocking service doesn't work         | Depends on the cloud infrastructure (ReadWriteMany)     |
+|                    | Less load on geoblocking service                        |                                                         |
+|                    | Works well with restart / upgrade / scaling             |                                                         |
+
+Decision: Let's use a Config Map as a fallback and cache for IP range allow/block list. However, let's split it technically from the Config Map that contains IP ranges provided by the end-user, so:
+- the Config Map containing custom IP range allow/block list is configured by the user, it becomes a contract
+- the Config Map containing IP range allow/block list downloaded from the SAP internal service is a Kyma internal thing and the implementation may change at any time, so no other module should use it
```

Comment:
There are more things to consider when going for ConfigMap as the backup/fallback:
- whether it should be accessible to be read (or should it be a secret)
- who should have access to this resource
- how to make sure that it is not being written to

Reply:
As we discussed - I've checked this and there is no difference in RBAC between ConfigMap and Secret. This CM would be stored in the same namespace where ip-auth is deployed, so we benefit from 'namespace separation' and Pods (Service Accounts) in other namespaces won't see it 'by default'. But a power user with access to kyma-system namespace may reconfigure it. So it is rather not possible to 'additionally' protect it, as the RBAC rules are additive and there is no possibility to 'deny' access to that resource if something has bigger permissions already.

So this comes down to:
- ip-auth having a Role that has a read/write access to this particular single CM (with IP rules cache) in kyma-system in kyma-system namespace
- ip-auth having a Role that has a read access to CM in kyma-system to be able to read CM with IP rules provided by the end-user
- remaining rules - we don't touch, if something has access to CM in kyma-system namespace or all namespaces (like admin user) - then they are able to access it

I'll add few words about it

Reply:
Added:
- a paragraph describing service accounts at the end of 'Namespaces' section
- additional sentence in 'IP range allow/block list fallback/caching' decision consequences